### PR TITLE
Keep Card/Page improvements

### PR DESCRIPTION
**Pull request by DerekBurch**
- Adds user/library chips to show libraries/users that are part of keepscussions
- Adds a keep page rhr behind an experiment
- Move the initial keep message/note into the comment section of keep cards
- Show a short activity attribution at the top of all keep cards

@andrewconner @cvializ

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/UserExperiment.scala [hunk 1]
```diff
@@ -80,6 +80,7 @@ object UserExperimentType {
   val SLACK = UserExperimentType("slack")
   val CUSTOM_LIBRARY_ORDERING = UserExperimentType("custom_library_ordering")
   val DISCUSSION_FEED_FILTERS = UserExperimentType("discussion_feed_filters")
+  val KEEP_PAGE_RHR = UserExperimentType("keep_page_rhr")
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/UserExperiment.scala:83]: Pro.

Comment (DerekBurch) [inline kifi-backend/modules/common/app/com/keepit/model/UserExperiment.scala:83]: Are these changes in UserExperiment all you need to do to add an experiment? I looked at an old commit and just copied what it was doing. I also didn't compile/test the Scala, but it seems okay.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/UserExperiment.scala:83]: Yep you good

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html [hunk 2]
```diff
@@ -161,11 +136,45 @@
...
-            ng-click="toggleExpandCard()"></span>
+      <div class="kf-keep-card-attribution-container">
+        <span ng-if="keep.discussion.numMessages || keep.displayNote"
+              class="kf-keep-card-view-comments"
+              ng-class="{'kf-link': !galleryView}"
+              ng-pluralize
+              ng-init="commentCount = keep.discussion.numMessages + (keep.displayNote ? 1 : 0)"
+              count="commentCount"
+              when="{'one': '1 comment', 'other': '{{commentCount|number}} comments'}"
+              ng-click="!galleryView && toggleExpandCard()"></span>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html:147]: do the double braces really work here? I would have expected `'other': (commentCount | number) + ' comments'`
but if it work it works.

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html [hunk 3]
```diff
@@ -161,11 +136,45 @@
...
+           ng-repeat="addedLibrary in keep.members.libraries"
+           ng-init="library = addedLibrary.library">
+          <svg kf-symbol-sprite
+               ng-if="library && library|libPrivacyIcon"
+               icon="{{::library|libPrivacyIcon}}"></svg>
+          <span ng-bind="library.name"></span>
+        </a>
+        <a class="kf-feed-tooltip-target kf-keep-card-participant-pic"
+           ng-href="{{::participant|profileUrl}}"
+           ng-attr-style="background-image:url({{::participant|pic:100}})"
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: using double braces partially like this breaks. use `ng-style` instead, e.g. `ng-style="::{'background-image': 'url(' + (participant | pic:100) + ')'}"`

**Comment by cvializ** [comment]
:sunglasses: :+1: :octopus:

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html (hunk 2 above)
Comment (DerekBurch) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html:147]: Seems like both ways work as far as I can tell and looks like what is going on in the docs https://docs.angularjs.org/api/ng/directive/ngPluralize.

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html (hunk 3 above)
Comment (DerekBurch) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: There are other uses of ng-attr-style in this file (some were there before these changes). Want me to change those as well? What is wrong with them? (As far as I can tell everything works).

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: I noticed at one point that they break in Internet Explorer, and it was difficult to track down to this usage. `ng-attr` also has some pretty weird semantics where it expects to receive an angular expression returning either a value to be used for success, or exactly `undefined` to remove the attribute altogether. `ng-style` doesn't have any of these weird expectations.

please do clean them up if you see them in a file you're working on. thanks : D

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html (hunk 2 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html:147]: ah you're right, carry on :bow:

File: kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html (hunk 3 above)
Comment (DerekBurch) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: Okay, their days are numbered.

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/keep/keepCard.tpl.html]: :dart: :gun:

**Pull request merged**